.github/workflows/ci.yml

```diff
@@ -2,42 +2,53 @@ name: CI
 
 on:
   push:
-    branches:
-      - main
+    branches: ["main"]
   pull_request:
-
-permissions:
-  actions: read
-  contents: read
+    types: [opened, synchronize]
 
 jobs:
-  main:
+  build:
+    name: Build, Test, and Lint
     runs-on: ubuntu-latest
+
     steps:
-      - uses: actions/checkout@v4
+      - name: Check out code
+        uses: actions/checkout@v4
         with:
-          fetch-depth: 0
-      
-      - uses: pnpm/action-setup@v2
+          fetch-depth: 2
+
+      - name: Cache turbo build setup
+        uses: actions/cache@v4
         with:
-          version: 8
-      
-      
+          path: .turbo
+          key: ${{ runner.os }}-turbo-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-turbo-
+
+      - name: Install build dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y libgtk-3-dev libwebkit2gtk-4.0-dev libappindicator3-dev librsvg2-dev patchelf libselinux1
 
-      # Connect your workspace on nx.app and uncomment this to enable task distribution.
-      # The "--stop-agents-after" is optional, but allows idle agents to shut down once the "build" targets have been requested
-      # - run: pnpm dlx nx-cloud start-ci-run --distribute-on="5 linux-medium-js" --stop-agents-after="build"
+      - name: Setup PNPM
+        uses: pnpm/action-setup@v2
+        with:
+          version: 9
```

Comment:
Since `package.json` specifies `packageManager`, I think it would be better to just replace this with `corepack enable` command (should be moved below `setup-node`), which will shim pnpm and automatically grab the appropriate version of it when it's run.

More information on corepack here:
- https://nodejs.org/api/corepack.html
- https://github.com/nodejs/corepack